File: 4139-minimum-absolute-distance-between-mirror-pairs/4139-minimum-absolute-distance-between-mirror-pairs.java
class Solution {
    public int minMirrorPairDistance(int[] nums) {
        int n = nums.length;
        int res = Integer.MAX_VALUE;
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = n - 1; i >= 0; i--) {
            int rev = getReverse(nums[i]);
            if (map.containsKey(rev)) {
                res = Math.min(res, Math.abs(i - map.get(rev)));
            }
            map.put(nums[i], i);
        }
        return res == Integer.MAX_VALUE ? -1 : res;
    }
    private int getReverse(int n) {
        int temp = n;
        int res = 0;
        while (temp > 0) {
            res = res * 10 + temp % 10;
            temp /= 10;
        }
        return res;
    }
}

File: 4139-minimum-absolute-distance-between-mirror-pairs/4139-minimum-absolute-distance-between-mirror-pairs.py
class Solution:
    def get_reverse(self, n: int) -> int:
        res = 0
        while n:
            res = res * 10 + n % 10
            n = n // 10
        return res
        
    def minMirrorPairDistance(self, nums: List[int]) -> int:
        n = len(nums)
        mp = dict()

        mini = 10 ** 9
        for i in range(n - 1, -1, -1):
            rev = self.get_reverse(nums[i])
            if rev in mp:
                mini = min(mini, abs(i - mp.get(rev))) 
            mp[nums[i]] = i

        if mini == 10 ** 9: return -1
        return mini
                
            
    
        